rethinkdb/datadog_checks/rethinkdb/backends.py

```diff
@@ -0,0 +1,93 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Callable, Iterator, List
+
+import rethinkdb
+
+from .config import Config
+from .connections import Connection, RethinkDBConnection
+from .exceptions import CouldNotConnect
+from .metrics.config import collect_config_totals
+from .metrics.current_issues import collect_current_issues
+from .metrics.statistics import (
+    collect_cluster_statistics,
+    collect_replica_statistics,
+    collect_server_statistics,
+    collect_table_statistics,
+)
+from .metrics.statuses import collect_server_status, collect_table_status
+from .metrics.system_jobs import collect_system_jobs
+from .queries import QueryEngine
+from .types import Metric
+from .version import parse_version
+
+
+class Backend(object):
```

Comment:
What exactly is the point of that class, considering you only have one implementation?

Reply:
No, I also have implementations in tests (although they don't derive from `Backend` from scratch):

https://github.com/DataDog/integrations-core/pull/5715/files#diff-982dcb0e148d94ed5ee681c16feebfdaR169-R176

So the point is a) better testability, and b) clearer separation of concerns (prevents the temptation to put everything on the check class, and binding the interface too much to the `rethinkdb` library).

Reply:
No I think the DefaultBackend class is great, I just don't see the point of having an interface for it.

Reply:
Yes, we don't even actually use the base `Backend` interface for tests, so I was tempted to ditch in the name of YAGNI too… I decided to keep it to clearly define (and document?) the contract, and avoid polluting it with implementation details due to us using the `rethinkdb` client library. Eg the stuff in the `DefaultBackend` constructor (`r`, `QueryEngine`) are implementation details.

Reply:
Interfaces helps separating contract from implementation, but I think it's not very pythonic. Feel a bit more Java way.

Just using private + public methods/attributes should be enough I think. With private methods/attributes prefixed with `_`, the rest is public.

Reply:
:+1:, simplified the backend implementation in 26e7dc9